### PR TITLE
[13524] React component

### LINKED ISSUE #13524: UI Component - React
This issue is to track the creation of the component: **React**

You can access the link at https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Mobile-Components?node-id=3161%3A67981.

Please have a chat with the design team before starting with the component, and don't forget to add sizing to the issue

### PR DESCRIPTION
**Pull request by ibrkhalil**
fixes #13524 

#### Platforms
- Android
- iOS

##### Functional
- 1-1 chats
- public chats
- group chats

status: ready
image
image

This is how it looks now, Latest look
image

### CODE REVIEW

**Comment by cammellos** [comment]
@ibrkhalil linting fails, you can run `make lint` locally to see the errors, any formatting can be fixed by running `make lint-fix`, but it will only address formatting issues

File: src/quo/previews/reacts.cljs [hunk 1]
```diff
@@ -0,0 +1,71 @@
+(ns quo.previews.reacts
```
Comment (Parveshdhull) [inline src/quo/previews/reacts.cljs]: Hi, Please use quo2 previews for redesign components

File: src/quo/previews/reacts.cljs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+                         :style (merge reaction-styling
+                                       {:margin-top 25
+                                        :background-color (if @is-open?
+                                                            bg-open
+                                                            bg-closed)})}
+   [icons/icon :main-icons/add-reaction-emoji
+    {:style {:color "black"
+             :opacity (if @is-open? 0.5 1)}}]])
+
+(defn preview-reacts []
```
Comment (Parveshdhull) [inline src/quo/previews/reacts.cljs]: We need to create a component inside quo2 and then create a preview using that component. 
Previews are just for testing and reviewing components, but we will use actual components to create new design screens later.
Also, I think one preview is enough for testing but the preview screen should provide controls to play/test that component.
Implementation example: https://github.com/status-im/status-react/pull/13597

File: src/quo2/screens/main.cljs [hunk 3]
```diff
@@ -4,22 +4,22 @@
...
 (def screens [{:name      :quo2-texts
                :insets    {:top false}
                :component text/preview-text}
+              {:name      :quo2-reacts
+               :insets    {:top false}
+               :component reacts/preview-reacts}
               {:name      :quo2-button
                :insets    {:top false}
                :component button/preview-button}
-              {:name      :quo2-status-tags
```
Comment (Parveshdhull) [inline src/quo2/screens/main.cljs]: Accidental removal?

**Review by Parveshdhull — CHANGES_REQUESTED** [review]

File: src/quo/previews/reacts.cljs (hunk 1 above)
Comment (ibrkhalil) [reply src/quo/previews/reacts.cljs]: Okay

File: src/quo/previews/reacts.cljs (hunk 2 above)
Comment (ibrkhalil) [reply src/quo/previews/reacts.cljs]: I see now
Will do it today

File: src/quo2/screens/main.cljs (hunk 3 above)
Comment (ibrkhalil) [reply src/quo2/screens/main.cljs]: Yes ..

File: src/quo/previews/reacts.cljs [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+(defn render-react
+  "Add your emoji as a param here"
+  [{:keys [emoji clicks dark? neutral?]}]
+  (let [text-color (if dark? "white" "black")
+        clicks-positive? (pos-int? @clicks)]
+    [rn/touchable-opacity {:on-press #(swap! clicks inc)
+                           :style (merge reaction-styling
+                                         {:border-color (if dark? "white" "black")
+                                          :background-color   (if dark?
+                                                                (if neutral? "#192438" "black")
```
Comment (cammellos) [inline src/quo/previews/reacts.cljs]: You can use the colors from `src/quo2/foundations/colors.cljs`, if you inspect in figma, you should see the name of the color and there should be a matching one in that file.
Also just by looking at the designs and the screenshots, seems like it's a bit off (for example there are no black borders on the designs, text color in dark mode is different), probably worth going over them again to check that they are matching, if you want we can go through them together and also @pedro-et can help out with how we use figma and how to find your way around components etc

Comment (ibrkhalil) [reply src/quo/previews/reacts.cljs]: Let's have a chat on it then today

File: src/quo2/screens/main.cljs (hunk 3 above)
Comment (flexsurfer) [reply src/quo2/screens/main.cljs]: could you revert please ?

**Comment by flexsurfer** [comment]
regarding icons naming https://discord.com/channels/624205794384281629/626744384565674034/991236399737098240

File: src/quo2/screens/main.cljs (hunk 3 above)
Comment (ibrkhalil) [reply src/quo2/screens/main.cljs]: Sure

File: ios/Podfile.lock [hunk 5]
```diff
@@ -665,6 +665,7 @@ SPEC CHECKSUMS:
   react-native-safe-area-context: 60f654e00b6cc416573f6d5dbfce3839958eb57a
   react-native-shake: de052eaa3eadc4a326b8ddd7ac80c06e8d84528c
   react-native-slider: 12bd76d3d568c9c5500825db54123d44b48e4ad4
+  react-native-splash-screen: 4312f786b13a81b5169ef346d76d33bc0c6dc457
```
Comment (flexsurfer) [inline ios/Podfile.lock]: seems not related to PR

File: src/quo2/screens/main.cljs [hunk 6]
```diff
@@ -15,10 +16,13 @@
...
+              {:name :quo2-status-tags
```
Comment (flexsurfer) [inline src/quo2/screens/main.cljs]: formatting

File: src/quo2/components/reacts.cljs [hunk 7]
```diff
@@ -0,0 +1,52 @@
+(ns quo2.components.reacts
+  (:require [quo.core :as quo]
+            [status-im.ui.components.icons.icons :as icons]
+            [quo.react-native :as rn]
+            [quo2.foundations.colors :as colors]))
+
+(def reaction-styling
+  {:display "flex"
```
Comment (flexsurfer) [inline src/quo2/components/reacts.cljs]: better to always use keywords instead of strings , like `:flex` , `:white` etc

File: src/quo2/components/reacts.cljs [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+    [rn/touchable-opacity {:on-press on-press
+                           :style (merge reaction-styling
+                                         {:margin-top 25
+                                          :border-width 1
+                                          :border-color   (if dark?
+                                                            colors/white-opa-5
+                                                            colors/neutral-80)})}
+     [icons/icon :main-icons/add-reaction-emoji
+      {:color (if dark?
+                "white"
```
Comment (flexsurfer) [inline src/quo2/components/reacts.cljs]: use white and black from colors ns

File: src/quo2/components/reacts.cljs [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+                "black")}]]))
+
+(defn render-react
+  "Add your emoji as a param here"
+  [{:keys [emoji clicks dark? neutral? on-press]}]
+  (let [text-color (if dark? "white" "black")
+        numeric-value (js/parseInt clicks)
```
Comment (flexsurfer) [inline src/quo2/components/reacts.cljs]: `int`

File: src/quo2/components/reacts.cljs [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+        clicks-positive? (pos-int? numeric-value)]
```
Comment (flexsurfer) [inline src/quo2/components/reacts.cljs]: `pos?`

**Review by flexsurfer — CHANGES_REQUESTED** [review]

File: src/quo2/screens/main.cljs (hunk 6 above)
Comment (ibrkhalil) [reply src/quo2/screens/main.cljs]: Fixed

File: src/quo2/components/reacts.cljs (hunk 7 above)
Comment (ibrkhalil) [reply src/quo2/components/reacts.cljs]: Done

File: src/quo2/components/reacts.cljs (hunk 8 above)
Comment (ibrkhalil) [reply src/quo2/components/reacts.cljs]: Fixed

File: src/quo2/components/reacts.cljs (hunk 9 above)
Comment (ibrkhalil) [reply src/quo2/components/reacts.cljs]: Done

File: src/quo2/components/reacts.cljs (hunk 10 above)
Comment (ibrkhalil) [reply src/quo2/components/reacts.cljs]: Fixed

File: src/quo2/components/reacts.cljs [hunk 11]
```diff
@@ -0,0 +1,52 @@
+(ns quo2.components.reacts
+  (:require [quo.core :as quo]
+            [status-im.ui.components.icons.icons :as icons]
+            [quo.react-native :as rn]
+            [quo2.foundations.colors :as colors]))
+
+(def reaction-styling
+  {:display :flex
+   :flex-direction "row"
```
Comment (flexsurfer) [inline src/quo2/components/reacts.cljs]: :row

**Review by flexsurfer — APPROVED** [review]
please format styles maps

File: src/quo2/components/reacts.cljs (hunk 11 above)
Comment (ibrkhalil) [reply src/quo2/components/reacts.cljs]: Done

File: src/quo2/components/reacts.cljs [hunk 12]
```diff
@@ -1,50 +1,66 @@
...
         clicks-positive? (pos? numeric-value)]
     [rn/touchable-opacity {:on-press on-press
                            :style (merge reaction-styling
                                          (cond-> {:background-color
                                                   (if dark?
-                                                    (if neutral? colors/neutral-70 colors/neutral-90)
-                                                    (if neutral? colors/neutral-30 colors/neutral-10))}
+                                                    (if neutral?
+                                                      colors/neutral-70
+                                                      "transparent")
```
Comment (flexsurfer) [inline src/quo2/components/reacts.cljs]: :transparent

File: src/quo2/components/reacts.cljs [hunk 13]
```diff
@@ -1,50 +1,66 @@
...
+                                                      "transparent"))}
                                            (and dark? (not neutral?)) (assoc :border-color colors/neutral-70
                                                                              :border-width 1)
                                            (and (not dark?) (not neutral?)) (assoc :border-color colors/neutral-30
                                                                                    :border-width 1)))}
-     [quo/text {:style {:color text-color}}
-      (str emoji (if clicks-positive?
-                   (str " " numeric-value)
-                   ""))]]))
+     [icons/icon emoji {:color "default"
```
Comment (flexsurfer) [inline src/quo2/components/reacts.cljs]: what is "default" color ?

Comment (ibrkhalil) [reply src/quo2/components/reacts.cljs]: I'll "nil" so that it uses the default color

File: src/quo2/components/reacts.cljs (hunk 12 above)
Comment (ibrkhalil) [reply src/quo2/components/reacts.cljs]: Done

File: src/quo2/components/reacts.cljs (hunk 13 above)
Comment (ibrkhalil) [reply src/quo2/components/reacts.cljs]: I just need to add the emoji's default color and not color it at all
If I didn't add the color attribute, it gets colored white

Comment (flexsurfer) [reply src/quo2/components/reacts.cljs]: `:no-color true`

File: src/quo2/components/reacts.cljs [hunk 14]
```diff
@@ -0,0 +1,66 @@
...
+                                                      colors/neutral-30
+                                                      :transparent))}
+                                           (and dark? (not neutral?)) (assoc :border-color colors/neutral-70
+                                                                             :border-width 1)
+                                           (and (not dark?) (not neutral?)) (assoc :border-color colors/neutral-30
+                                                                                   :border-width 1)))}
+     [icons/icon emoji {:color "nil"
+                        :width 16
+                        :height 16}]
+     [quo/text {:style {:font-size 13
```
Comment (siddarthkay) [inline src/quo2/components/reacts.cljs]: please use `quo2/text` instead of `quo/text` which is part of legacy design system and will be deprecated soon

File: src/quo2/components/reacts.cljs [hunk 15]
```diff
@@ -0,0 +1,66 @@
...
+            [status-im.ui.components.icons.icons :as icons]))
+
+(def reaction-styling
+  {:flex-direction :row
+   :justify-content :center
+   :align-items :center
+   :padding-vertical 3
+   :padding-horizontal 8
+   :border-radius 8
+   :gap 4})
```
Comment (siddarthkay) [inline src/quo2/components/reacts.cljs]: I believe there is no `:gap` property in style can you double check once?

**Review by siddarthkay — CHANGES_REQUESTED** [review]

**Comment by siddarthkay** [comment]
@ibrkhalil : Design issue 1 

The add emoji icon/button seems to have a very dark border which is not how it is seen in figma. Its either the border width that is off or the border color.

Screenshot 2022-08-22 at 6 08 47 PM

File: src/quo2/components/reacts.cljs (hunk 15 above)
Comment (ibrkhalil) [reply src/quo2/components/reacts.cljs]: Removed, Thanks Sid!

File: src/quo2/components/reacts.cljs (hunk 13 above)
Comment (ibrkhalil) [reply src/quo2/components/reacts.cljs]: Thanks Andre
Done

File: src/quo2/components/reacts.cljs (hunk 14 above)
Comment (ibrkhalil) [reply src/quo2/components/reacts.cljs]: Done

File: src/quo2/components/reacts.cljs [hunk 16]
```diff
@@ -0,0 +1,65 @@
...
+                                                      colors/neutral-30
+                                                      :transparent))}
+                                           (and dark? (not neutral?)) (assoc :border-color colors/neutral-70
+                                                                             :border-width 1)
+                                           (and (not dark?) (not neutral?)) (assoc :border-color colors/neutral-30
+                                                                                   :border-width 1)))}
+     [icons/icon emoji {:no-color true
+                        :width 16
+                        :height 16}]
+     [quo2-text/text {:style {:font-size 13
```
Comment (siddarthkay) [inline src/quo2/components/reacts.cljs]: For `font-weight` and `font-size` please use typography namespace instead.
I believe quo2/text already accepts these keys as part of weight and size parameters.

here is an example for your reference : 
https://github.com/status-im/status-mobile/blob/fcb2a2d553758ad0165918b9252c326b47e64d04/src/quo2/components/account_selector.cljs#L69

**Review by siddarthkay — CHANGES_REQUESTED** [review]

File: src/quo2/components/reacts.cljs (hunk 16 above)
Comment (ibrkhalil) [reply src/quo2/components/reacts.cljs]: Done

**Review by siddarthkay — APPROVED** [review]

**Review by Parveshdhull — APPROVED** [review]
